src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -2270,7 +2270,7 @@ public bool CheckIfIdAlreadyIncluded(string[] ids, IEnumerable<string> includes)
             {
                 var id = result.GetMetadata().GetId();
                 if (idsEntitiesPairs.TryGetValue(id, out var tuple) == false)
-                    ThrowCouldNotRefreshDocument($"Could not refresh a entity with id: {id}");
+                    ThrowCouldNotRefreshDocument($"Cannot refresh a transient instance with document id: {id}");
```

Comment:
Wait, this is _after_ we called, no? So this isn't a trasient instance.
And how are you detecting that the document was deleted here? You are iterating over the results from the server here, no?
So if there is no document, you'll get null?

Reply:
I will phrase another message to describe why. 
for your second question: yes, we will get null. 
trying to refresh `{d1,d2}` when d1 was deleted will result with null in BlittableJsonReaderArray `command.Result.Results` json.
`{[null, {"Name": "d1Name", "@metadata": {...}}]}`